app/components/windows/SourceFilters.vue

```diff
@@ -43,7 +43,7 @@
 
       <div class="modal-container--side-nav">
         <div v-if="selectedFilterName">
-          <GenericForm v-model="properties" @input="save"></GenericForm>
+          <GenericForm v-model="properties" @input="save" :key="selectedFilterName"></GenericForm>
```

Comment:
This is an extra level of security.  It forces us to remake the entire form when selecting a different filter, at the expense of some performance when swapping filters.